fatiando/gravmag/harvester.py

```diff
@@ -579,10 +579,13 @@ def _test_restriction(restrict):
     """
     Test for correct spelling of items in restrict list.
     """
-    cases = ['above', 'below', 'north', 'south', 'east', 'west']
-    for case in restrict:
-        if not(case in cases):
-            raise ValueError("Unrecognized item in restrict: %s" % case)
+    cases = ['above', 'below', 'north', 'south', 'east', 'west', None]
+    if type(restrict) is list:
+        for case in restrict:
+            if not(case in cases):
+                raise ValueError("Unrecognized item in restrict: %s" % case)
+    else:
+        raise TypeError("Argument restrict is not a list.")
```

Comment:
Since `restrict=None` by default (see above), this check can be simpler:

    if restrict is None:
        return []
    else:
        for case in restrict:
            if case not in cases:
                raise ValueError(...)
    return restrict

Then there is no need to the type check and the code will work just as fine if `restrict` is some other iterable type, like a tuple or set.

In Python, it's generally best to avoid doing `type` checks. Instead, execute the code and it will break if you did something that isn't supported, like using an integer in a `for` loop.

Notice as well that the function now parses the `restrict` and returns the proper form (`[]` if it was `None`). Now you can put `restrict = _test_restrict(restrict)` in the main `harvester` code and remove the check if `restrict is None` from `_neighbor_indexes`.